src/com/ms/silverking/cloud/skfs/meta/SKFSConfigurationZK.java

```diff
@@ -11,43 +11,43 @@
 import com.ms.silverking.cloud.management.MetaToolOptions;
 import com.ms.silverking.io.StreamParser;
 
-public class SKFSConfigurationZK extends MetaToolModuleBase<SKFSConfiguration,MetaPaths> {
-    public SKFSConfigurationZK(MetaClient mc) throws KeeperException {
-        super(mc, mc.getMetaPaths().getConfigPath());
-    }
-    
-    @Override
-    public SKFSConfiguration readFromFile(File file, long version) throws IOException {
-        com.ms.silverking.cloud.skfs.meta.MetaClient skfsMc = (com.ms.silverking.cloud.skfs.meta.MetaClient) (this.mc);
-        return new SKFSConfiguration( skfsMc.getSKFSConfigName(), StreamParser.parseLines(file), version, 0L);
-        
-    }
+public class SKFSConfigurationZK extends MetaToolModuleBase<SKFSConfiguration, MetaPaths> {
+  public SKFSConfigurationZK(MetaClient mc) throws KeeperException {
+    super(mc, mc.getMetaPaths().getConfigPath());
+  }
 
-    @Override
-    public SKFSConfiguration readFromZK(long version, MetaToolOptions options) throws KeeperException {
-        com.ms.silverking.cloud.skfs.meta.MetaClient skfsMc = (com.ms.silverking.cloud.skfs.meta.MetaClient) (this.mc);
-        return SKFSConfiguration.parse(skfsMc.getSKFSConfigName(), zk.getString(getVBase(version)), version);
-    }
-    
-    @Override
-    public void writeToFile(File file, SKFSConfiguration instance) throws IOException {
-        if (file != null) {
-            FileOutputStream fs ;
-            fs = new FileOutputStream(file);
-            fs.write(instance.toString().getBytes());
-            fs.flush();
-            fs.close();
-        }
-        else {
-            System.out.println(instance.toString());
-        }
+  @Override
+  public SKFSConfiguration readFromFile(File file, long version) throws IOException {
+    com.ms.silverking.cloud.skfs.meta.MetaClient skfsMc = (com.ms.silverking.cloud.skfs.meta.MetaClient) (this.mc);
+    return new SKFSConfiguration(skfsMc.getSKFSConfigName(), StreamParser.parseLines(file), version, 0L);
+
+  }
+
+  @Override
+  public SKFSConfiguration readFromZK(long version, MetaToolOptions options) throws KeeperException {
+    com.ms.silverking.cloud.skfs.meta.MetaClient skfsMc = (com.ms.silverking.cloud.skfs.meta.MetaClient) (this.mc);
+    return SKFSConfiguration.parse(skfsMc.getSKFSConfigName(), zk.getString(getVBase(version)), version);
+  }
+
+  @Override
+  public void writeToFile(File file, SKFSConfiguration instance) throws IOException {
+    if (file != null) {
+      FileOutputStream fs;
+      fs = new FileOutputStream(file);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *fs*. The resource is referenced in statements at the following lines: 37, 38. The resource closure statement is at line: 39. There are other execution paths that do not contain closure statements, for example, when FileOutputStream.write throws an exception. Either a) close *fs* in a try-finally block or b) close the resource by declaring *fs* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).